server/src/lib.rs

```diff
@@ -214,26 +231,35 @@ const MAX_CLIENT_MSG_SIZE: usize = 1 << 16;
 
 async fn drive_recv(
     id: ClientId,
-    mut streams: quinn::IncomingUniStreams,
+    connection: quinn::Connection,
     send: &mut mpsc::Sender<(ClientId, ClientEvent)>,
 ) -> Result<()> {
-    let hello = match streams.next().await {
-        None => return Ok(()),
-        Some(stream) => {
-            codec::recv_whole::<proto::ClientHello>(MAX_CLIENT_MSG_SIZE, stream?).await?
-        }
-    };
+    let stream = connection.accept_uni().await.map_err(Error::msg)?;
+    let hello = codec::recv_whole::<proto::ClientHello>(MAX_CLIENT_MSG_SIZE, stream).await?;
     let _ = send.send((id, ClientEvent::Hello(hello))).await;
 
-    let mut cmds = streams
-        .map(|stream| async {
-            codec::recv_whole::<proto::Command>(MAX_CLIENT_MSG_SIZE, stream?).await
-        })
-        .buffer_unordered(16); // Allow a modest amount of out-of-order completion
-    while let Some(msg) = cmds.try_next().await? {
-        let _ = send.send((id, ClientEvent::Command(msg))).await;
+    loop {
+        let stream = connection.accept_uni().await.map_err(Error::msg)?;
+        let send = send.clone();
+
+        // We spawn a separate task to allow messages to be processed in a different order from when they were
+        // initiated.
+        let connection = connection.clone();
+        tokio::spawn(async move {
+            match codec::recv_whole::<proto::Command>(MAX_CLIENT_MSG_SIZE, stream).await {
+                Err(e) => {
+                    // This error can occur if the client sends a badly-formatted command. In this case,
+                    // we want to drop the client. We close the connection, which will cause `drive_recv` to
+                    // return eventually.
+                    tracing::error!("Error when parsing unordered stream from server: {e}");
```

Comment:
Should this read "from client"?

Reply:
It should. Thanks! I copied this from the version I wrote in `net.rs`.